nodes/server.js

```diff
@@ -17,14 +18,17 @@ module.exports = function (RED) {
             node.refreshFindInterval = node.config.polling * 1000;
             node.on('close', () => this.onClose());
 
-            node.connect().then(result => {
-                node.getStatus(true).then(result => {
-                    node.emit("onInitEnd", result);
-                });
-            });
+            retryOperation(node.connect.bind(node), 'Connecting Miio Roborock')
+                .then((result) => {
+                    return node.getStatus(true)
+                        .then((result) => {
+                            node.emit("onInitEnd", result);
+                        });
+                })
+                .catch((e) => console.log('Connecting to Miio Roborock failed', e));
```

Comment:
By default retrying 10000 times with 5s intervals. That means it retries about up to 14 hours and then finally fails and logs out this message.